### PR TITLE
Test release testing.

### PR DESCRIPTION
**Pull request by granluo**
Create release testing by updating Podfile to make SpecStaging repo as the source of pods.

### CODE REVIEW

**Comment by granluo** [comment]
Test is done in 2e50955.

File: scripts/localize_podfile.swift [hunk 1]
```diff
@@ -65,14 +76,18 @@ for line in lines {
         didImplicits = true
         for implicit in implicitPods {
           let implicitPodspec = repo.appendingPathComponent(implicit + ".podspec").path
-          outBuffer += "pod '\(implicit)', :path => '\(implicitPodspec)'\n"
+          outBuffer +=
+            releaseTesting
+            ? "pod '\(implicit)'\n" : "pod '\(implicit)', :path => '\(implicitPodspec)'\n"
```
Comment (paulb777) [inline scripts/localize_podfile.swift:81]: Implicit pods should not be used at all for release testing.

If all the podspecs are pushed to staging, they will be found from the `source` directive in the Podfile.

I can do a GVC tomorrow if you want to discuss verbally.

File: .github/workflows/remoteconfig.yml [hunk 2]
```diff
@@ -90,6 +90,34 @@ jobs:
     - name: Test Swift Quickstart
       run: ([ -z $plist_secret ] || scripts/third_party/travis/retry.sh scripts/test_quickstart.sh Config)
 
+  quickstart_release_test:
```
Comment (paulb777) [inline .github/workflows/remoteconfig.yml:93]: I'm not sure about putting this in remoteconfig.yml versus a separate release.yml for all of them.

File: scripts/localize_podfile.swift (hunk 1 above)
Comment (granluo) [reply scripts/localize_podfile.swift:81]: Sure we can do the GVC. This change will set specstaging as the source of Podfile only and will not update SpecStaging repo. I am planning to update the Podfile first to let podfile find in the repo first and then we can determine how to update the SpecStaging.
I thought implict pods were involved in the tests as well since they are in the release podspec repo, but we can discuss which pods should be tested in the meeting .

File: .github/workflows/remoteconfig.yml (hunk 2 above)
Comment (granluo) [reply .github/workflows/remoteconfig.yml:93]: I prefer a separate release.yml since 
1. badges on readme will be very specific to the latest pods in repo. The result is intuitive to developers.
2. If release testings need to be triggered, we don't need to trigger nightly quickstart testing.

File: scripts/localize_podfile.swift [hunk 3]
```diff
@@ -61,18 +72,20 @@ for line in lines {
     }
     let podspec = repo.appendingPathComponent(podName + ".podspec").path
     if FileManager().fileExists(atPath: podspec) {
-      if didImplicits == false {
+      if didImplicits == false, !releaseTesting {
```
Comment (granluo) [inline scripts/localize_podfile.swift:75]: Implicit pods will not be added if it is release testing.

File: scripts/localize_podfile.swift [hunk 4]
```diff
@@ -45,11 +55,12 @@ while url.path != "/", url.lastPathComponent != "firebase-ios-sdk" {
...
-  "source 'https://cdn.cocoapods.org/'\n"
+var outBuffer =
+  "source 'https://github.com/firebase/SpecsStaging.git'\n"
+    + "source 'https://cdn.cocoapods.org/'\n"
 for line in lines {
-  var newLine = line
-  let tokens = line.components(separatedBy: [" ", ","] as CharacterSet)
+  var newLine = line.trimmingCharacters(in: .whitespacesAndNewlines)
+  let tokens = newLine.components(separatedBy: [" ", ","] as CharacterSet)
   if tokens.first == "pod" {
```
Comment (paulb777) [inline scripts/localize_podfile.swift:64]: I think this can be `if tokens.first == "pod" && !releaseTesting {` and then undo the changes inside the test.

The only change necessary for testing the staged pods should be adding source `'https://github.com/firebase/SpecsStaging.git'` to the Podfile.
